Makefile

```diff
@@ -1,96 +1,79 @@
-
-.PHONY: clean
-clean:
-	rm -rf dist || true
-	cd $(LIBBPF_SRC) && $(MAKE) clean;
-	rm -f $(BPF_OBJ)
-	rm -f *.ll
-	rm -rf .output
+$(OUT_BIN): $(BPF_OBJ) $(LIBBPF_OBJ) $(GO_SRC) | $(OUT_DIR)
+	GOOS=linux GOARCH=$(ARCH:x86_64=amd64) \
+		CGO_CFLAGS="-I $(abspath $(LIBBPF_HEADERS))/bpf" \
+		CGO_LDFLAGS="$(abspath $(LIBBPF_OBJ))" \
+		go build -v -o $(OUT_BIN) \
+		-ldflags "-X github.com/aquasecurity/tracee/tracee.ebpfProgramB64Injected=$$(base64 -w 0 $(BPF_SRC))"
 
-imageName ?= tracee
-.PHONY: docker
-docker:
-	docker build -t $(imageName) .
+$(OUT_BIN)_slim: $(BPF_OBJ) $(LIBBPF_OBJ) $(GO_SRC) | $(OUT_DIR)
+	GOOS=linux GOARCH=$(ARCH:x86_64=amd64) go build -v -o $(OUT_BIN)
 
-.PHONY: release
-# release by default will not publish. run with `publish=1` to publish
-goreleaserFlags = --skip-publish --snapshot
-ifdef publish
-	goreleaserFlags =
-endif
-release:
-	EBPFPROGRAM_BASE64=$(ebpfProgramBase64) goreleaser release --rm-dist $(goreleaserFlags)
+# .PHONY: build-docker
+# build-docker: clean
+# 	img=$$(docker build --target builder -q  .) && \
+# 	cnt=$$(docker create $$img) && \
+# 	docker cp $$cnt:/tracee/dist - | tar -xf - ; \
+# 	docker rm $$cnt ; docker rmi $$img
 
-.PHONY: $(CLANG) $(LLC)
+bpf_compile_tools = $(LLC) $(CLANG) $(LLVM_STRIP)
+.PHONY: $(bpf_compile_tools) 
+$(bpf_compile_tools): % : check_%
 
-llvm-check: $(CLANG) $(LLC)
-	@for TOOL in $^ ; do \
-		if [ ! $$(command -v $${TOOL} 2>/dev/null) ]; then \
-			echo "*** ERROR: Cannot find tool $${TOOL}" ;\
-			exit 1; \
-		else true; fi; \
-	done
+$(LIBBPF_SRC):
+	test -d $(LIBBPF_SRC) || ( echo "missing libbpf source, try git submodule update --init" ; false )
 
-$(OUTPUT):
-	mkdir -p $@
+$(LIBBPF_HEADERS): | $(OUT_DIR) $(bpf_compile_tools) $(LIBBPF_SRC)
+	cd $(LIBBPF_SRC) && $(MAKE) install_headers DESTDIR=$(abspath $(OUT_DIR))/libbpf 
 
-$(LIBBPF_OBJ):
-	@if [ ! -d $(LIBBPF_SRC) ]; then \
-		echo "Error: Need libbpf submodule"; \
-		echo "May need to run git submodule update --init"; \
-		exit 1; \
-	else \
-		cd $(LIBBPF_SRC) && $(MAKE) BUILD_STATIC_ONLY=1; \
-		OBJDIR=$(dir $@)/libbpf DESTDIR=$(dir $@) $(MAKE) install_headers; \
-	fi
+$(LIBBPF_OBJ): | $(OUT_DIR) $(bpf_compile_tools) $(LIBBPF_SRC) 
+	cd $(LIBBPF_SRC) && $(MAKE) OBJDIR=$(abspath $(OUT_DIR))/libbpf BUILD_STATIC_ONLY=1 
 
-$(BPF_OBJ): ${BPF_C} $(LIBBPF_OBJ) | ${OUTPUT}
+linux_arch := $(ARCH:x86_64=x86)
+bpf_extra_headers := 3rdparty/include #copy to out?
+$(BPF_OBJ): $(BPF_SRC) $(LIBBPF_HEADERS) | $(OUT_DIR) $(bpf_compile_tools)
 	$(CLANG) -S \
-		-D __BPF_TRACING__ -D __KERNEL__ -D__TARGET_ARCH_$(ARCH) \
-		$(CFLAGS) \
+		-D __BPF_TRACING__ -D __KERNEL__ -D__TARGET_ARCH_$(linux_arch) \
+		-I $(LIBBPF_HEADERS) \
 		-include $(KERN_SRC)/include/linux/kconfig.h \
-		-I $(KERN_SRC)/arch/$(ARCH)/include \
-		-I $(KERN_SRC)/arch/$(ARCH)/include/uapi \
-		-I $(KERN_SRC)/arch/$(ARCH)/include/generated \
-		-I $(KERN_SRC)/arch/$(ARCH)/include/generated/uapi \
+		-I $(KERN_SRC)/arch/$(linux_arch)/include \
```

Comment:
Here, linux_arch should be arm64 (and not aarch64 given by uname -m) for arm64 architecture